### PR TITLE
Archive users/members from the members page

### PR DESCRIPTION
**Pull request by MehulKChaudhari**
This PR solves issue #240 

This PR should be merged only after merging PR #260

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,5 +56,8 @@
   },
   "engines": {
     "node": "14.x"
+  },
+  "optionalDependencies": {
```
[the remark this answers is not in the document]
Comment (MehulKChaudhari) [reply package.json]: Removed the dependency, resolved the error another way.

File: src/pages/index.js [hunk 2]
```diff
@@ -64,7 +64,10 @@ export async function getServerSideProps(context) {
     membersArr.sort((a, b) => (a.first_name > b.first_name ? 1 : -1));
 
     const newMembersArr = membersArray.filter(
-      (person) => !person.isMember && person.first_name !== undefined
+      (person) =>
+        !person.isMember &&
+        person.first_name !== undefined &&
+        !person.archived_member
```
Comment (ankushdharkar) [inline src/pages/index.js:70]: Why are we not keeping the field names consistent with camelCasing?

File: src/helper-functions/urls.js [hunk 3]
```diff
@@ -43,6 +43,9 @@ const getActiveTasksURL = (rdsId) => `${baseURL}/tasks/${rdsId}?status=active`;
 const getAddMemberRoleURL = (rdsId) =>
   `${baseURL}/members/moveToMembers/${rdsId}`;
 
+const getArchiveMemberUrl = (rdsId) =>
```
Comment (ankushdharkar) [inline src/helper-functions/urls.js:46]: Why is it URL above, but Url here?

File: src/components/member-role-update/index.js [hunk 4]
```diff
@@ -26,6 +26,16 @@ const MemberRoleUpdate = () => {
       setUpdateStatus('Some error occured, please contact admin');
     }
   };
+  const archiveMember = async (user) => {
+    setIsUpdating(true);
+    const { status } = await archiveMember(user);
+    setIsUpdating(false);
```
Comment (ankushdharkar) [inline src/components/member-role-update/index.js:32]: Should we imperatively set the spinner logic, or have API state derive this data?

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: src/helper-functions/urls.js (hunk 3 above)
Comment (MehulKChaudhari) [reply src/helper-functions/urls.js:46]: Yeah, it should be a `URL `to keep code consistent.

File: src/pages/index.js (hunk 2 above)
Comment (MehulKChaudhari) [reply src/pages/index.js:70]: I didn't paid much attention, Kill keep in mind from now.

File: src/components/member-role-update/index.js (hunk 4 above)
Comment (MehulKChaudhari) [reply src/components/member-role-update/index.js:32]: Yes we can do it API state derived, and that makes more sense.

Comment (MehulKChaudhari) [reply src/components/member-role-update/index.js:32]: https://github.com/Real-Dev-Squad/website-members/issues/272

**Review by ankushdharkar — APPROVED** [review]

**Pull request merged**